src/OrchardCore.Modules/OrchardCore.Queries/Sql/Drivers/SqlQueryDisplayDriver.cs

```diff
@@ -40,18 +40,16 @@ public override IDisplayResult Edit(SqlQuery query, IUpdateModel updater)
                 // Extract query from the query string if we come from the main query editor
                 if (string.IsNullOrEmpty(query.Template))
                 {
-                    updater.TryUpdateModelAsync(model, "", m => m.Query);
-
-                    // The value is empty for new queries, remove any errors
-                    if (string.IsNullOrEmpty(model.Query))
-                    {
-                        updater.ModelState.Clear();
-                    }
+                    var queryStringModel = new QueryStringModel();
```

Comment:
Now I remember what this code is for. When from the query editor we click on create. Have you tested that and found it was not working? I'll try it too.

Reply:
Yes, i tried it. Here we try to update the `Query` from the query string if coming from the query editor. Then if `Query` is null that would mean we create a new query, **so we clear the errors**. But this could also happen after a regular query publish / save, so after executing the below `UpdateAsync()`, and when the `Query` field was left empty.

The result was that the required `Query` field was no more checked even it is left empty because errors were cleared.